lib/file.js

```diff
@@ -206,54 +205,54 @@ class File extends EventEmitter {
 
-      let activeStreams = 0
-      let currentOffset = apiStart
-      let chunkSize = initialChunkSize
-      let combined
-
-      function handleErrors (resp) {
-        if (resp.statusCode === 200) return
+      function handleMegaErrors (resp) {
+        if (resp.statusCode === 200) return;
         stream.emit('error', resp.statusCode === 509
         ? Error('Bandwidth limit reached: ' + resp.headers['x-mega-time-left'] + ' seconds until it resets')
-        : Error('MEGA returned a ' + resp.statusCode + ' status code'))
-        gotError = true
+        : Error('MEGA returned a ' + resp.statusCode + ' status code'));
       }
-
-      function getChunk () {
-        if (gotError) return
-        const currentMax = Math.min(end, currentOffset + chunkSize)
-        if (currentMax <= currentOffset) return
-        const r = requestModule(response.g + '/' + currentOffset + '-' + (currentMax - 1))
-
-        r.on('end', getChunk)
-        r.on('error', getChunk)
-        r.on('response', handleErrors)
-
-        combined.append(r)
-
-        currentOffset = currentMax
-        if (chunkSize < maxChunkSize) {
-          chunkSize = chunkSize + chunkSizeIncrement
-        }
-
-        activeStreams += 1
-        if (activeStreams < maxConnections) {
-          setTimeout(getChunk, 1000)
-        }
+      
+      function handleConnectionErrors(resp) {
+          stream.emit('error', Error("Connection error: " + resp.message));
       }
 
       if (maxConnections === 1) {
-        const r = requestModule(response.g + '/' + apiStart + '-' + (end - 1))
-        r.on('error', handleErrors)
-        r.pipe(decryptStream)
+        const r = requestModule(response.g + '/' + apiStart + '-' + (end - 1));
+        r.on('error', handleConnectionErrors);
+        r.on('response', handleMegaErrors);
+        r.pipe(decryptStream);
       } else {
-        combined = CombinedStream.create()
-
+        let currentOffset = apiStart,
+            chunkSize = initialChunkSize,
+            combined = CombinedStream.create(),
+            gotError = false
+        ;
+        stream.on('error', () => {gotError = true});
+
+        let getChunk = function () {
+            if (gotError) return;
+            const currentMax = Math.min(end, currentOffset + chunkSize);
+            if (currentMax <= currentOffset) return;
+            const r = requestModule(response.g + '/' + currentOffset + '-' + (currentMax - 1));
+
+            r.on('end', getChunk);
+            r.on('error', handleConnectionErrors);
+            r.on('response', handleMegaErrors);
+
+            combined.append(r);
+
+            currentOffset = currentMax;
+            if (chunkSize < maxChunkSize) {
+                chunkSize = chunkSize + chunkSizeIncrement
+            }
+        };
         // Pass errors from the combined stream to the main stream
-        combined.on('error', (err) => stream.emit('error', err))
+        combined.on('error', (err) => stream.emit('error', err));
```

Comment:
[Follow the code style](http://standardjs.com/). I put linting in the "dist" script, not in the "test" one: I will fix that.